File: blackduck-artifactory-common/src/test/groovy/com/synopsys/integration/blackduck/artifactory/GeneralPropertyTest.java
package com.synopsys.integration.blackduck.artifactory;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.EnumSource;
import org.junit.platform.commons.util.StringUtils;

import com.synopsys.integration.blackduck.artifactory.configuration.GeneralProperty;
import com.synopsys.integration.blackduck.artifactory.util.Fast;

class GeneralPropertyTest {
    @ParameterizedTest
    @Fast
    @EnumSource(GeneralProperty.class)
    void getKey(final GeneralProperty property) {
        assertNotNull(property.getKey());
        assertTrue(StringUtils.isNotBlank(property.getKey()));
    }
}

File: blackduck-artifactory-common/src/test/kotlin/com/synopsys/integration/blackduck/artifactory/configuration/GeneralPropertyTest.kt
/**
 * blackduck-artifactory-common
 *
 * Copyright (c) 2021 Synopsys, Inc.
 *
 * Use subject to the terms and conditions of the Synopsys End User Software License and Maintenance Agreement. All rights reserved worldwide.
 */
package com.synopsys.integration.blackduck.artifactory.configuration

import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Assertions.assertTrue

import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.EnumSource
import org.junit.platform.commons.util.StringUtils

class GeneralPropertyTest {
    @ParameterizedTest
    @EnumSource(GeneralProperty::class)
    fun getKey(property: GeneralProperty) {
        assertNotNull(property.key, "A GeneralProperty must have a key.")
        assertTrue(StringUtils.isNotBlank(property.key), "A GeneralProperty key must not be blank.")
    }
}